From 04e4bd619b533a6cc317843cf251052348050444 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sat, 10 Mar 2018 14:10:16 +0100
Subject: [PATCH 5/8] nxp: jni: Remove unused variables and functions

 * nxp/jni/com_android_nfc_NativeLlcpSocket.cpp:
    28:13: error: unused function 'nfc_jni_disconnect_callback'

 * nxp/jni/com_android_nfc_NativeLlcpConnectionlessSocket.cpp:
    61:20: error: unused variable 'ts'
    123:20: error: unused variable 'ts'

 * nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp:
    74:20: error: unused variable 'ts'

 * nxp/jni/com_android_nfc_NativeLlcpSocket.cpp:
    115:20: error: unused variable 'ts'
    169:20: error: unused variable 'ts'
    251:20: error: unused variable 'ts'
    312:20: error: unused variable 'ts'

 * nxp/jni/com_android_nfc_NativeNfcManager.cpp:
    135:13: error: unused variable 'InputBuffer'
    134:13: error: unused variable 'OutputBuffer'
    237:10: error: unused variable 'value'
    298:20: error: unused variable 'ts'
    303:64: error: unused variable 'SmartMX_detected'
    306:12: error: unused variable 'firmware_status'
    303:45: error: unused variable 'SmartMX_index'
    732:29: error: unused variable 'nfc_jni_nfcip1_cfg'
    904:14: error: unused variable 'ret'
    905:20: error: unused variable 'ts'
    908:17: error: unused variable 'typeName'
    1107:13: error: unused function 'nfc_jni_smartMX_setModeCb'
    1122:27: error: unused parameter 'evt_info'
    1128:13: error: unused variable 'mifare_block'
    1130:20: error: unused variable 'aid'
    1131:20: error: unused variable 'mifare_command'
    1132:35: error: unused variable 'pCallbackData'
    1133:9: error: unused variable 'i'
    1350:71: error: unused parameter 'restart'
    1576:14: error: unused variable 'status'
    1579:12: error: unused variable 'obj'
    1640:13: error: unused variable 'result'
    1687:1: error: unused label 'clean_and_return'

 * nxp/jni/com_android_nfc_NativeNfcTag.cpp:
    27:24: error: unused variable 'handle'
    119:13: error: unused function 'nfc_jni_async_disconnect_callback'
    120:46: error: unused parameter 'status'
    716:11: error: unused variable 'technologies'
    710:9: error: unused variable 'res'

 * nxp/jni/com_android_nfc_NativeP2pDevice.cpp:
    34:13: error: unused function 'nfc_jni_presence_check_callback'
    128:13: error: unused variable 'tag'
    129:15: error: unused variable 'ctor'
    274:12: error: unused variable 'offset'
    360:20: error: unused variable 'ts'

Change-Id: I75f7459bb65d4f4a983f95c60a7bd98400f62837
---
 ...oid_nfc_NativeLlcpConnectionlessSocket.cpp |  2 -
 ...om_android_nfc_NativeLlcpServiceSocket.cpp |  1 -
 nxp/jni/com_android_nfc_NativeLlcpSocket.cpp  | 20 ---------
 nxp/jni/com_android_nfc_NativeNfcManager.cpp  | 44 +++----------------
 nxp/jni/com_android_nfc_NativeNfcTag.cpp      | 18 +-------
 nxp/jni/com_android_nfc_NativeP2pDevice.cpp   | 17 -------
 6 files changed, 7 insertions(+), 95 deletions(-)

diff --git a/nxp/jni/com_android_nfc_NativeLlcpConnectionlessSocket.cpp b/nxp/jni/com_android_nfc_NativeLlcpConnectionlessSocket.cpp
index b1fe791e..de1237cf 100644
--- a/nxp/jni/com_android_nfc_NativeLlcpConnectionlessSocket.cpp
+++ b/nxp/jni/com_android_nfc_NativeLlcpConnectionlessSocket.cpp
@@ -58,7 +58,6 @@ static void nfc_jni_send_callback(void *pContext, NFCSTATUS status)
 static jboolean com_android_nfc_NativeLlcpConnectionlessSocket_doSendTo(JNIEnv *e, jobject o, jint nsap, jbyteArray data)
 {
    NFCSTATUS ret;
-   struct timespec ts;
    phLibNfc_Handle hRemoteDevice;
    phLibNfc_Handle hLlcpSocket;
    phNfc_sData_t sSendBuffer = {NULL, 0};
@@ -120,7 +119,6 @@ clean_and_return:
 static jobject com_android_nfc_NativeLlcpConnectionlessSocket_doReceiveFrom(JNIEnv *e, jobject o, jint linkMiu)
 {
    NFCSTATUS ret;
-   struct timespec ts;
    uint8_t ssap;
    jobject llcpPacket = NULL;
    phLibNfc_Handle hRemoteDevice;
diff --git a/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp b/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp
index a2f57978..9b4bd76f 100644
--- a/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp
+++ b/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp
@@ -71,7 +71,6 @@ static phLibNfc_Handle getIncomingSocket(nfc_jni_native_monitor_t * pMonitor,
 static jobject com_NativeLlcpServiceSocket_doAccept(JNIEnv *e, jobject o, jint miu, jint rw, jint linearBufferLength)
 {
    NFCSTATUS ret = NFCSTATUS_SUCCESS;
-   struct timespec ts;
    phLibNfc_Llcp_sSocketOptions_t sOptions;
    phNfc_sData_t sWorkingBuffer;
    jfieldID f;   
diff --git a/nxp/jni/com_android_nfc_NativeLlcpSocket.cpp b/nxp/jni/com_android_nfc_NativeLlcpSocket.cpp
index 6a0fda28..2d087393 100644
--- a/nxp/jni/com_android_nfc_NativeLlcpSocket.cpp
+++ b/nxp/jni/com_android_nfc_NativeLlcpSocket.cpp
@@ -21,22 +21,6 @@
 
 namespace android {
 
-/*
- * Callbacks
- */
- 
-static void nfc_jni_disconnect_callback(void*        pContext,
-                                               NFCSTATUS    status)
-{
-   struct nfc_jni_callback_data * pCallbackData = (struct nfc_jni_callback_data *) pContext;
-   LOG_CALLBACK("nfc_jni_disconnect_callback", status);
-
-   /* Report the callback status and wake up the caller */
-   pCallbackData->status = status;
-   sem_post(&pCallbackData->sem);
-}
-
- 
 static void nfc_jni_connect_callback(void* pContext, uint8_t nErrCode, NFCSTATUS status)
 {
    struct nfc_jni_callback_data * pCallbackData = (struct nfc_jni_callback_data *) pContext;
@@ -112,7 +96,6 @@ static void nfc_jni_send_callback(void *pContext, NFCSTATUS status)
 static jboolean com_android_nfc_NativeLlcpSocket_doConnect(JNIEnv *e, jobject o, jint nSap)
 {
    NFCSTATUS ret;
-   struct timespec ts;
    phLibNfc_Handle hRemoteDevice;
    phLibNfc_Handle hLlcpSocket;
    struct nfc_jni_callback_data cb_data;
@@ -166,7 +149,6 @@ clean_and_return:
 static jboolean com_android_nfc_NativeLlcpSocket_doConnectBy(JNIEnv *e, jobject o, jstring sn)
 {
    NFCSTATUS ret;
-   struct timespec ts;
    phNfc_sData_t serviceName = {NULL, 0};
    phLibNfc_Handle hRemoteDevice;
    phLibNfc_Handle hLlcpSocket;
@@ -248,7 +230,6 @@ static jboolean com_android_nfc_NativeLlcpSocket_doClose(JNIEnv *e, jobject o)
 static jboolean com_android_nfc_NativeLlcpSocket_doSend(JNIEnv *e, jobject o, jbyteArray  data)
 {
    NFCSTATUS ret;
-   struct timespec ts;
    phLibNfc_Handle hRemoteDevice;
    phLibNfc_Handle hLlcpSocket;
    phNfc_sData_t sSendBuffer = {NULL, 0};
@@ -309,7 +290,6 @@ clean_and_return:
 static jint com_android_nfc_NativeLlcpSocket_doReceive(JNIEnv *e, jobject o, jbyteArray  buffer)
 {
    NFCSTATUS ret;
-   struct timespec ts;
    phLibNfc_Handle hRemoteDevice;
    phLibNfc_Handle hLlcpSocket;
    phNfc_sData_t sReceiveBuffer = {NULL, 0};
diff --git a/nxp/jni/com_android_nfc_NativeNfcManager.cpp b/nxp/jni/com_android_nfc_NativeNfcManager.cpp
index f611f3e5..fc5de458 100644
--- a/nxp/jni/com_android_nfc_NativeNfcManager.cpp
+++ b/nxp/jni/com_android_nfc_NativeNfcManager.cpp
@@ -131,8 +131,6 @@ static void nfc_jni_deinit_download_callback(void *pContext, NFCSTATUS status)
 
 static int nfc_jni_download_locked(struct nfc_jni_native_data *nat, uint8_t update)
 {
-    uint8_t OutputBuffer[1];
-    uint8_t InputBuffer[1];
     struct timespec ts;
     NFCSTATUS status = NFCSTATUS_FAILED;
     phLibNfc_StackCapabilities_t caps;
@@ -236,7 +234,6 @@ clean_and_return:
 
 static int nfc_jni_configure_driver(struct nfc_jni_native_data *nat)
 {
-    char value[PROPERTY_VALUE_MAX];
     int result = FALSE;
     NFCSTATUS status;
 
@@ -297,15 +294,13 @@ static int nfc_jni_unconfigure_driver(struct nfc_jni_native_data* /*nat*/)
 
 /* Initialization function */
 static int nfc_jni_initialize(struct nfc_jni_native_data *nat) {
-   struct timespec ts;
    uint8_t resp[16];
    NFCSTATUS status;
    phLibNfc_StackCapabilities_t caps;
    phLibNfc_SE_List_t SE_List[PHLIBNFC_MAXNO_OF_SE];
-   uint8_t i, No_SE = PHLIBNFC_MAXNO_OF_SE, SmartMX_index = 0, SmartMX_detected = 0;
+   uint8_t i, No_SE = PHLIBNFC_MAXNO_OF_SE;
    phLibNfc_Llcp_sLinkParameters_t LlcpConfigInfo;
    struct nfc_jni_callback_data cb_data;
-   uint8_t firmware_status;
    uint8_t update = TRUE;
    int result = JNI_FALSE;
    const hw_module_t* hw_module;
@@ -734,12 +729,6 @@ extern uint8_t nfc_jni_is_ndef;
 extern uint8_t *nfc_jni_ndef_buf;
 extern uint32_t nfc_jni_ndef_buf_len;
 
-static phLibNfc_sNfcIPCfg_t nfc_jni_nfcip1_cfg =
-{
-   3,
-   { 0x46, 0x66, 0x6D }
-};
-
 /*
  * Callbacks
  */
@@ -906,9 +895,6 @@ static void nfc_jni_Discovery_notification_callback(void *pContext,
    phLibNfc_RemoteDevList_t *psRemoteDevList,
    uint8_t uNofRemoteDev, NFCSTATUS status)
 {
-   NFCSTATUS ret;
-   const char * typeName;
-   struct timespec ts;
    phNfc_sData_t data;
    int i;
    int target_index = 0; // Target that will be reported (if multiple can be >0)
@@ -1111,32 +1097,14 @@ static void nfc_jni_deinit_callback(void *pContext, NFCSTATUS status)
    sem_post(&pContextData->sem);
 }
 
-/* Set Secure Element mode callback*/
-static void nfc_jni_smartMX_setModeCb (void*            pContext,
-                                       phLibNfc_Handle  /*hSecureElement*/,
-                                       NFCSTATUS        status)
-{
-   struct nfc_jni_callback_data * pContextData =  (struct nfc_jni_callback_data*)pContext;
-
-   LOG_CALLBACK("nfc_jni_smartMX_setModeCb", status);
-
-   pContextData->status = status;
-   sem_post(&pContextData->sem);
-}
-
 /* Card Emulation callback */
 static void nfc_jni_transaction_callback(void *context,
    phLibNfc_eSE_EvtType_t evt_type, phLibNfc_Handle /*handle*/,
-   phLibNfc_uSeEvtInfo_t *evt_info, NFCSTATUS status)
+   phLibNfc_uSeEvtInfo_t *evt_info __unused, NFCSTATUS status)
 {
     JNIEnv *e;
     jobject tmp_array = NULL;
-    jobject mifare_block = NULL;
     struct nfc_jni_native_data *nat;
-    phNfc_sData_t *aid;
-    phNfc_sData_t *mifare_command;
-    struct nfc_jni_callback_data *pCallbackData;
-    int i=0;
 
     LOG_CALLBACK("nfc_jni_transaction_callback", status);
 
@@ -1360,7 +1328,8 @@ static void com_android_nfc_NfcManager_disableDiscovery(JNIEnv *e, jobject o)
 
 // TODO: use enable_lptd
 static void com_android_nfc_NfcManager_enableDiscovery(JNIEnv *e, jobject o, jint modes,
-        jboolean, jboolean reader_mode, jboolean enable_p2p, jboolean restart)
+        jboolean, jboolean reader_mode, jboolean enable_p2p,
+        jboolean restart __unused)
 {
     NFCSTATUS ret;
     struct nfc_jni_native_data *nat;
@@ -1586,10 +1555,8 @@ static jint com_android_nfc_NfcManager_doGetTimeout(JNIEnv*, jobject,
 
 static jboolean com_android_nfc_NfcManager_init_native_struc(JNIEnv *e, jobject o)
 {
-   NFCSTATUS status;
    struct nfc_jni_native_data *nat = NULL;
    jclass cls;
-   jobject obj;
    jfieldID f;
 
    TRACE("******  Init Native Structure ******");
@@ -1650,7 +1617,6 @@ static jboolean com_android_nfc_NfcManager_initialize(JNIEnv *e, jobject o)
 #ifdef TNFC_EMULATOR_ONLY
    char value[PROPERTY_VALUE_MAX];
 #endif
-   jboolean result;
 
    CONCURRENCY_LOCK();
 
@@ -1697,7 +1663,9 @@ static jboolean com_android_nfc_NfcManager_initialize(JNIEnv *e, jobject o)
    /* Perform the initialization */
    init_result = nfc_jni_initialize(nat);
 
+#ifdef TNFC_EMULATOR_ONLY
 clean_and_return:
+#endif
    CONCURRENCY_UNLOCK();
 
    /* Convert the result and return */
diff --git a/nxp/jni/com_android_nfc_NativeNfcTag.cpp b/nxp/jni/com_android_nfc_NativeNfcTag.cpp
index 932b8410..c4557b14 100644
--- a/nxp/jni/com_android_nfc_NativeNfcTag.cpp
+++ b/nxp/jni/com_android_nfc_NativeNfcTag.cpp
@@ -24,7 +24,6 @@
 #include "phNfcHalTypes.h"
 
 static phLibNfc_Data_t nfc_jni_ndef_rw;
-static phLibNfc_Handle handle;
 uint8_t *nfc_jni_ndef_buf = NULL;
 uint32_t nfc_jni_ndef_buf_len = 0;
 
@@ -99,7 +98,7 @@ static void nfc_jni_checkndef_callback(void *pContext,
 }
 
 static void nfc_jni_disconnect_callback(void *pContext,
-   phLibNfc_Handle /*hRemoteDev*/, NFCSTATUS status)
+   phLibNfc_Handle /*hRemoteDev*/, NFCSTATUS status __unused)
 {
    struct nfc_jni_callback_data * pCallbackData = (struct nfc_jni_callback_data *) pContext;
    LOG_CALLBACK("nfc_jni_disconnect_callback", status);
@@ -116,19 +115,6 @@ static void nfc_jni_disconnect_callback(void *pContext,
    sem_post(&pCallbackData->sem);
 }
 
-static void nfc_jni_async_disconnect_callback(void * /*pContext*/,
-   phLibNfc_Handle /*hRemoteDev*/, NFCSTATUS status)
-{
-   LOG_CALLBACK("nfc_jni_async_disconnect_callback", status);
-
-   if(nfc_jni_ndef_buf)
-   {
-      free(nfc_jni_ndef_buf);
-   }
-   nfc_jni_ndef_buf = NULL;
-   nfc_jni_ndef_buf_len = 0;
-}
-
 static phNfc_sData_t *nfc_jni_transceive_buffer;
 
 static void nfc_jni_transceive_callback(void *pContext,
@@ -707,13 +693,11 @@ static jbyteArray com_android_nfc_NativeNfcTag_doTransceive(JNIEnv *e,
     uint32_t outlen;
     phLibNfc_sTransceiveInfo_t transceive_info;
     jbyteArray result = NULL;
-    int res;
     phLibNfc_Handle handle = nfc_jni_get_connected_handle(e, o);
     NFCSTATUS status;
     struct nfc_jni_callback_data cb_data;
     int selectedTech = 0;
     int selectedLibNfcType = 0;
-    jint* technologies = NULL;
     bool checkResponseCrc = false;
 
     jint *targetLost;
diff --git a/nxp/jni/com_android_nfc_NativeP2pDevice.cpp b/nxp/jni/com_android_nfc_NativeP2pDevice.cpp
index 28bd8c00..6acb8f80 100644
--- a/nxp/jni/com_android_nfc_NativeP2pDevice.cpp
+++ b/nxp/jni/com_android_nfc_NativeP2pDevice.cpp
@@ -27,19 +27,6 @@ extern uint8_t device_connected_flag;
 namespace android {
 
 extern void nfc_jni_restart_discovery_locked(struct nfc_jni_native_data *nat);
-
-/*
- * Callbacks
- */
-static void nfc_jni_presence_check_callback(void* pContext, NFCSTATUS status)
-{   
-   struct nfc_jni_callback_data * pCallbackData = (struct nfc_jni_callback_data *) pContext;
-   LOG_CALLBACK("nfc_jni_presence_check_callback", status);
-
-   /* Report the callback status and wake up the caller */
-   pCallbackData->status = status;
-   sem_post(&pCallbackData->sem);
-}
  
 static void nfc_jni_connect_callback(void *pContext,
                                      phLibNfc_Handle /*hRemoteDev*/,
@@ -125,8 +112,6 @@ static jboolean com_android_nfc_NativeP2pDevice_doConnect(JNIEnv *e, jobject o)
     struct nfc_jni_callback_data cb_data;
 
     ScopedLocalRef<jclass> target_cls(e, NULL);
-    jobject tag;
-    jmethodID ctor;
     jfieldID f;
     jbyteArray generalBytes = NULL;
     phNfc_sData_t sGeneralBytes;
@@ -271,7 +256,6 @@ static jbyteArray com_android_nfc_NativeP2pDevice_doTransceive(JNIEnv *e,
    jobject o, jbyteArray data)
 {
    NFCSTATUS status;
-   uint8_t offset = 2;
    uint8_t *buf;
    uint32_t buflen;
    phLibNfc_sTransceiveInfo_t transceive_info;
@@ -357,7 +341,6 @@ static jbyteArray com_android_nfc_NativeP2pDevice_doReceive(
    JNIEnv *e, jobject o)
 {
    NFCSTATUS status;
-   struct timespec ts;
    phLibNfc_Handle handle;
    jbyteArray buf = NULL;
    static phNfc_sData_t *data;
-- 
2.17.1

